scripts/migration/migrate_backrefs_between_nodes_and_nodelogs.py

```diff
@@ -0,0 +1,70 @@
+"""
+This migration will add original_node and node associated with the log to nodelogs. It will then
+clone each nodelog for the remaining nodes in the backref (registrations and forks),
+changing the node to the current node.
+"""
+from copy import deepcopy
+import logging
+import sys
+
+from framework.mongo import database as db
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+
+def migrate(dry=True):
+    init_app(routes=False)
+    node_logs = db.nodelog.find({'original_node': None})
+    total_log_count = node_logs.count()
+    count = 0
+
+    for log in node_logs:
+        count += 1
+        try:
+            # Step 1: migrate original nodelog - updates existing
+            db.nodelog.update(
+                {'_id': log['_id']},
+                {'$set': {
+                    'original_node': log['params']['node'],
+                    'node': log['__backrefs']['logged']['node']['logs'][0]
+                    }
+                },
+                upsert=True
+            )
+            logger.info('{}/{} Log {} updated'.format(count, total_log_count, log['_id']))
+        except KeyError as error:
+            logger.error('Could not migrate nodelog due to error -- likely a lack of __backrefs')
```

Comment:
I don't know the best way to handle this.  It works fine on local data - but there might be broken logs in production/staging.  There should be a nodes on a log's backrefs - could the logs without backrefs be recorded as this script is running and then those logs handled individually?